libraries/AP_Proximity/AP_Proximity_AirSimSITL.cpp

```diff
@@ -0,0 +1,83 @@
+#include <AP_HAL/AP_HAL.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+#include "AP_Proximity_AirSimSITL.h"
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define PROXIMITY_MAX_RANGE 100.0f
+#define PROXIMITY_ACCURACY 0.1f
+
+/* 
+   The constructor also initialises the proximity sensor. 
+*/
+AP_Proximity_AirSimSITL::AP_Proximity_AirSimSITL(AP_Proximity &_frontend,
+                                     AP_Proximity::Proximity_State &_state):
+    AP_Proximity_Backend(_frontend, _state),
+    sitl(AP::sitl())
+{
+}
+
+// update the state of the sensor
+void AP_Proximity_AirSimSITL::update(void)
+{
+    SITL::vector3f_array &points = sitl->state.scanner.points;
+    if (points.length == 0) {
+        set_status(AP_Proximity::Proximity_NoData);
+        return;
+    }
+
+    set_status(AP_Proximity::Proximity_Good);
+
+    memset(_distance_valid, 0, sizeof(_distance_valid));
+    memset(_angle, 0, sizeof(_angle));
+    memset(_distance, 0, sizeof(_distance));
+
+    // only use 8 sectors to match RPLidar
+    const uint8_t nsectors = MIN(8, PROXIMITY_SECTORS_MAX);
+    const uint16_t degrees_per_sector = 360 / nsectors;
+
+    for (uint16_t i=0; i<points.length; i++) {
+        Vector3f &point = points.data[i];
+        if (point.is_zero()) {
+            continue;
+        }
+        float angle_deg = wrap_360(degrees(atan2f(-point.y, point.x)));
+        uint16_t angle_rounded = uint16_t(angle_deg+0.5);
+        uint8_t sector = wrap_360(angle_rounded + 22.5f) / degrees_per_sector;
+        if (!_distance_valid[sector] || PROXIMITY_MAX_RANGE < _distance[sector]) {
+            _distance_valid[sector] = true;
+            const Vector2f v = Vector2f(point.x, point.y);
+            _distance[sector] = v.length();
+            _angle[sector] = angle_deg;
+            update_boundary_for_sector(sector, true);
```

Comment:
Bool field was added after recent rebase on master, `true` since it was the same on other backends
But haven't tested if this changes anything!